File: CS/DXSample/Metadata.cs
using DevExpress.Mvvm.DataAnnotations;
using System;

namespace DXSample {
    public class Metadata : IMetadataProvider<Customer> {
        void IMetadataProvider<Customer>.BuildMetadata(MetadataBuilder<Customer> builder) {
            builder
                .Property(x => x.ID)
                    .ReadOnly()
                .EndProperty()
                .Property(x => x.FirstName)
                    .DisplayName("First Name")
                    .MaxLength(40)
                .EndProperty()
                .Property(x => x.LastName)
                    .DisplayName("Last Name")
                    .MaxLength(40)
                .EndProperty()
                .Property(x => x.FullName)
                    .Hidden()
                .EndProperty()
                .Property(x => x.Phone)
                    .DisplayName("Phone number")
                    .PhoneNumberDataType()
                .EndProperty()
                .Property(x => x.Email)
                    .DisplayName("Email")
                    .EmailAddressDataType()
                .EndProperty()
                .Property(x => x.BirthDate)
                    .DisplayName("Birth date")
                    .DateTimeDataType()
                .EndProperty();
            builder
                .Group("Personal")
                    .ContainsProperty(x => x.ID)
                    .ContainsProperty(x => x.BirthDate)
                .EndGroup()
                .Group("Name")
                    .ContainsProperty(x => x.FirstName)
                    .ContainsProperty(x => x.LastName)
                .EndGroup()
                .Group("Contact")
                    .ContainsProperty(x => x.Phone)
                    .ContainsProperty(x => x.Email)
                .EndGroup();
        }
    }

    public class ReviewMetadata : IMetadataProvider<Customer> {
        void IMetadataProvider<Customer>.BuildMetadata(MetadataBuilder<Customer> builder) {
            builder
                .Property(x => x.ID)
                    .Hidden()
                .EndProperty()
                .Property(x => x.FirstName)
                    .Hidden()
                .EndProperty()
                .Property(x => x.LastName)
                    .Hidden()
                .EndProperty()
                .Property(x => x.FullName)
                    .DisplayName("Full Name")
                    .ReadOnly()
                .EndProperty()
                .Property(x => x.Phone)
                    .DisplayName("Phone number")
                    .PhoneNumberDataType()
                .EndProperty()
                .Property(x => x.Email)
                    .DisplayName("Email")
                    .EmailAddressDataType()
                .EndProperty()
                .Property(x => x.BirthDate)
                    .Hidden()
                .EndProperty();
            builder
                .Group("Name")
                    .ContainsProperty(x => x.FullName)
                .EndGroup()
                .Group("Contact")
                    .ContainsProperty(x => x.Phone)
                    .ContainsProperty(x => x.Email)
                .EndGroup();
        }
    }
}

File: VB/DXSample/Metadata.vb
Imports DevExpress.Mvvm.DataAnnotations

Namespace DXSample

    Public Class Metadata
        Implements IMetadataProvider(Of Customer)

        Private Sub BuildMetadata(ByVal builder As MetadataBuilder(Of Customer)) Implements IMetadataProvider(Of Customer).BuildMetadata
            PropertyMetadataBuilderExtensions.DateTimeDataType(PropertyMetadataBuilderExtensions.EmailAddressDataType(PropertyMetadataBuilderExtensions.PhoneNumberDataType(builder.Property(Function(x) x.ID).ReadOnly().EndProperty().Property(Function(x) x.FirstName).DisplayName("First Name").MaxLength(40).EndProperty().Property(Function(x) x.LastName).DisplayName("Last Name").MaxLength(40).EndProperty().Property(Function(x) x.FullName).Hidden().EndProperty().Property(Function(x) x.Phone).DisplayName("Phone number")).EndProperty().Property(Function(x) x.Email).DisplayName("Email")).EndProperty().Property(Function(x) x.BirthDate).DisplayName("Birth date")).EndProperty()
            builder.Group("Personal").ContainsProperty(Function(x) x.ID).ContainsProperty(Function(x) x.BirthDate).EndGroup().Group("Name").ContainsProperty(Function(x) x.FirstName).ContainsProperty(Function(x) x.LastName).EndGroup().Group("Contact").ContainsProperty(Function(x) x.Phone).ContainsProperty(Function(x) x.Email).EndGroup()
        End Sub
    End Class

    Public Class ReviewMetadata
        Implements IMetadataProvider(Of Customer)

        Private Sub BuildMetadata(ByVal builder As MetadataBuilder(Of Customer)) Implements IMetadataProvider(Of Customer).BuildMetadata
            PropertyMetadataBuilderExtensions.EmailAddressDataType(PropertyMetadataBuilderExtensions.PhoneNumberDataType(builder.Property(Function(x) x.ID).Hidden().EndProperty().Property(Function(x) x.FirstName).Hidden().EndProperty().Property(Function(x) x.LastName).Hidden().EndProperty().Property(Function(x) x.FullName).DisplayName("Full Name").ReadOnly().EndProperty().Property(Function(x) x.Phone).DisplayName("Phone number")).EndProperty().Property(Function(x) x.Email).DisplayName("Email")).EndProperty().Property(Function(x) x.BirthDate).Hidden().EndProperty()
            builder.Group("Name").ContainsProperty(Function(x) x.FullName).EndGroup().Group("Contact").ContainsProperty(Function(x) x.Phone).ContainsProperty(Function(x) x.Email).EndGroup()
        End Sub
    End Class
End Namespace
